server/api/crud/runtimes/nuclio/function.py

```diff
@@ -293,10 +296,16 @@ def _compile_function_config(
         )
         # since environment variables have a limited size,
         # large serving specs are stored in config maps that are mounted to the pod
+        serving_spec_len = len(serving_spec)
         if (
             can_pass_via_cm
-            and len(serving_spec) >= mlrun.mlconf.httpdb.nuclio.serving_spec_env_cutoff
+            and serving_spec_len >= mlrun.mlconf.httpdb.nuclio.serving_spec_env_cutoff
         ):
+            if serving_spec_len >= SERVING_SPEC_MAX_LENGTH:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"Serving spec length {serving_spec_len} is too large. "
+                    f"The maximum allowed length for the serving spec is {SERVING_SPEC_MAX_LENGTH}."
```

Comment:
I agree with @TomerShor that we should change this error message to something the user can work with. Possibly point to `mlrun.runtimes.nuclio.serving.ServingRuntime._get_serving_spec` and tell the user to check it to see what takes so much space. Other than that lgtm

Reply:
Error messages will be defined by @jillnogold in this sub-document https://iguazio.atlassian.net/browse/ML-7694